content/en/docs/components/trainer/user-guides/data-cache.md

```diff
@@ -0,0 +1,161 @@
++++
+title = "Distributed Data Cache"
+description = "How to use distributed data cache with Kubeflow Trainer"
+weight = 40
++++
+
+This guide describes how to use distributed data cache to stream data into distributed TrainJobs.
+
+## Overview
+
+The data cache feature enables efficient data streaming for distributed training workloads by:
+
+- Pre-processing and caching data in a distributed Arrow cache cluster
+- Streaming data directly to training nodes without redundant preprocessing
+- Enabling scalable data access across multiple training nodes
+
+Data cache automatically fetches data from object store and partitions it across data nodes:
+
+<img src="/docs/components/trainer/images/data-cache-batch.png"
+  alt="Data Cache Batch"
+  class="mt-3 mb-3 border rounded p-3 bg-white"
+  style="width: 100%; max-width: 30em">
+
+Multiple TrainJob can access data from the cache using Apache Arrow flight protocol:
+
+<img src="/docs/components/trainer/images/data-cache-trainjob-access.png"
+  alt="Data Cache TrainJob"
+  class="mt-3 mb-3 border rounded p-3 bg-white"
+  style="width: 100%; max-width: 30em">
+
+The data cache is powered by [Apache Arrow](https://arrow.apache.org/) and [Apache DataFusion](https://datafusion.apache.org/)
+to effectively store data in-memory
+with zero-copy transfer to GPU nodes.
+
+## Architecture
+
+The training workflow consists of two stages:
+
+1. **Dataset Initializer**: Sets up a distributed cache cluster that preprocesses and serves the training data
+2. **Training Nodes**: Stream data from the cache and perform model training
+
+<img src="/docs/components/trainer/images/data-cache-init.png"
+  alt="Data Cache Init"
+  class="mt-3 mb-3 border rounded p-3 bg-white"
+  style="width: 100%; max-width: 30em">
+
+## Prerequisites
+
+Before exploring this guide, make sure to follow [the Getting Started guide](/docs/components/trainer/getting-started/)
+to understand the basics of Kubeflow Trainer and install controller manager.
+
+### Install Data Cache Control Plane
+
+You need to install the following resources to use data cache:
+
+- LeaderWorkerSet controller manager
+- ClusterTrainingRuntime with cache support: `torch-distributed-with-cache`
+- RBAC resources needed for initializer to bootstrap cache
+
+Run the following command to install the required resources:
+
+```bash
+kubectl apply --server-side -k "https://github.com/kubeflow/trainer.git/manifests/overlays/data-cache?ref=master"
+```
+
+{{% alert title="Note" color="info" %}}
+
+The above command will install RBAC in the `default` namespace. If you want to create TrainJobs
+in other Kubernetes namespace, update the RBAC resources and manually deploy it in your namespace:
+
+```bash
+https://github.com/kubeflow/trainer/tree/master/manifests/base/runtimes/data_cache/rbac.yaml
```

Comment:
This can be updated to `kubectl apply -k manifests/base/runtimes/data_cache -n <namespace> --server-side` to install rbac in other namespaces